crates/forge/src/running.rs

```diff
@@ -291,12 +303,15 @@ fn get_fork_state_reader(
     }
 }
 
-fn get_latest_block_number(url: &Url) -> Result<BlockId> {
+fn get_latest_block_number(
+    url: &Url,
+) -> Result<BlockId, ProviderError<JsonRpcClientError<HttpTransportError>>> {
     let client = JsonRpcClient::new(HttpTransport::new(url.clone()));
     let runtime = Runtime::new().expect("Could not instantiate Runtime");
 
     match runtime.block_on(client.get_block_with_tx_hashes(BlockId::Tag(Latest))) {
         Ok(MaybePendingBlockWithTxHashes::Block(block)) => Ok(BlockId::Number(block.block_number)),
-        _ => Err(anyhow!("Could not get the latest block number".to_string())),
+        Ok(MaybePendingBlockWithTxHashes::PendingBlock(_)) => unreachable!(),
```

Comment:
This is out of scope

Reply:
It is not, before we returned a general error that would kill the whole runner. Now I want to return a specific error that occurs on rpc call, so I have to add this variant to the match statement to be able to get the error arm below. It is impossible for us to ever get `PendingBlock` here.